src/unix/linux_like/linux/mod.rs

```diff
@@ -6129,6 +6144,28 @@ extern "C" {
     pub fn ioctl(fd: ::c_int, request: ::Ioctl, ...) -> ::c_int;
 }
 
+cfg_if! {
+    if #[cfg(target_env = "musl")] {
+        extern "C" {
+            pub fn getitimer(which: ::c_int, value: *mut ::itimerval) -> ::c_int;
```

Comment:
not sure I follow, why not using the typedef you created above ?

Reply:
This is the result of a strange combination of issues. glibc and most other libc implementations typedef `__itimer_which_t` as an `enum` when `_GNU_SOURCE` is set, or as an `int` otherwise. `musl` unfortunately opts to just use an `int` directly within the function definitions for `getitimer` and `setitimer`. As a result of this:

- Using `__itimer_whiich_t` directly in function definitions causes `musl` CI to fail as it's not a defined type
- Removing `__itimer_which_t` altogether for `musl` builds still causes CI to fail, as there is currently no way of opting out of `enum` semver checks conditionally for `musl` (you can opt out for `struct`s, `type`s, everything else but `enum`s)

By defining `__itimer_which_t` as a `type` definition in `musl` and then opting out of it, CI passes.

A more clean solution to this may be to develop `enum` opt-out functions to the `ctest2` crate.

Reply:
Another possible approach is not to bother with enums at all.

```rust
pub type __itimer_which_t = ::c_int;
...
pub const ITIMER_REAL: __itimer_which_t = 0;
...
```

There are various places in this crate were C enums are handled as simple constants. Should pass in theory.